app/models/important_notice.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: important_notices
+#
+#  id                    :bigint           not null, primary key
+#  can_dismiss           :boolean          default(FALSE)
+#  date_time             :datetime
+#  dismissed_at          :datetime
+#  message               :text
+#  notice_type           :integer          not null
+#  created_at            :datetime         not null
+#  updated_at            :datetime         not null
+#  dismissed_by_user_id  :bigint
+#  patient_id            :bigint           not null
+#  team_id               :bigint           not null
+#  vaccination_record_id :bigint
+#
+# Indexes
+#
+#  index_important_notices_on_dismissed_by_user_id          (dismissed_by_user_id)
+#  index_important_notices_on_patient_id                    (patient_id)
+#  index_important_notices_on_team_id                       (team_id)
+#  index_important_notices_on_vaccination_record_id         (vaccination_record_id)
+#  index_notices_on_patient_and_type_and_datetime_and_team  (patient_id,notice_type,date_time,team_id) UNIQUE
+#
+# Foreign Keys
+#
+#  fk_rails_...  (dismissed_by_user_id => users.id)
+#  fk_rails_...  (patient_id => patients.id)
+#  fk_rails_...  (team_id => teams.id)
+#  fk_rails_...  (vaccination_record_id => vaccination_records.id)
+#
+class ImportantNotice < ApplicationRecord
+  belongs_to :patient
+
+  enum :notice_type,
```

Comment:
```suggestion
  enum :type,
```

We would typically not put the model name as a prefix for this, you can avoid the issue around single table inheritance by setting `self.inheritance_column = nil`.

See https://github.com/nhsuk/manage-vaccinations-in-schools/blob/d7b5972927002900c3f66e7ab79f0d31b7d75431/app/models/parent_relationship.rb#L36 for an example.